### ISSUE TITLE
Add support for configuring overlay VXLAN options

### ISSUE THREAD

**Issue by gaby** [moby/moby#49143]
### Description

The current overlay driver uses VXLAN but does not allow specifying any VXLAN options like:
- vni
- multicast group

Other network drivers are quite configurable, the overlay one seems to be restricted on that aspect.

TLDR: I have a VXLAN interface on host A (docker not installed) that wants to communicate with containers on an overlay network on host B (w/ docker).

My local interface is create as:

```bash
ip link add vxlan100 type vxlan id 100 group 239.1.1.1 dev eth0 dstport 4789
ip addr add 10.0.0.2/24 dev vxlan100
ip link set up dev vxlan100
```

**Comment by thompson-shaun** [comment]
@akerouanton and @robmry

**Comment by akerouanton** [comment]
Mirantis has a few employees maintaining Swarm and the overlay driver. So let me ping @aepifanov and @corhere.

**Comment by gaby** [comment]
@akerouanton Any updates?

**Comment by corhere** [comment]
The VNI is already configurable, though how to do so is possibly undocumented. [Why would you want to, though?](https://xyproblem.info) The overlay network driver is not interoperable with other VXLAN implementations as it uses a custom control plane for neighbor advertisement and discovery. Is it a problem of coexistence with other VXLANs, where the VNIs automatically chosen by the overlay driver collide with VNIs for other non-Docker VXLANs on the same host?

The multicast group is not configurable because the overlay driver does not currently support multicasting of VXLAN datagrams.

**Comment by corhere** [comment]
> TLDR: I have a VXLAN interface on host A (docker not installed) that wants to communicate with containers on an overlay network on host B (w/ docker).

@gaby what you want to do is not possible. Things that are not Docker containers cannot (successfully) communicate with containers on an overlay network. Overlay networks utilize VXLAN encapsulation, but also require a large userspace component to maintain the forwarding database that tells the kernel which tunnel endpoint to forward each encapsulated packet onto. The overlay driver is active in the background throughout the lifetime of the container doing control-plane things to keep packets flowing as neighbors connect and disconnect from the overlay network.

The macvlan or ipvlan drivers might be a better fit for your use-case of mixing container and non-container endpoints on a virtual network segment.

**Comment by gaby** [comment]
> The VNI is already configurable, though how to do so is possibly undocumented. [Why would you want to, though?](https://xyproblem.info) The overlay network driver is not interoperable with other VXLAN implementations as it uses a custom control plane for neighbor advertisement and discovery. Is it a problem of coexistence with other VXLANs, where the VNIs automatically chosen by the overlay driver collide with VNIs for other non-Docker VXLANs on the same host?
> 
> The multicast group is not configurable because the overlay driver does not currently support multicasting of VXLAN datagrams.

I believe the lack of multicast would be fixed by https://github.com/moby/moby/pull/49733 ?

**Comment by corhere** [comment]
@gaby no, I do not believe so. If I understand your request correctly, you want the option to use VXLAN in multicast mode, where VXLAN UDP datagrams may be multicast across the physical network to all the other nodes listening to the same multicast group. In contrast, #49733 is an attempt to support multicasting inside containers connected to overlay networks. With the implementation in that PR, copies of the VXLAN UDP datagrams carrying multicast payloads (addressed to any multicast group) would themselves be unicast to all the other nodes.

**Comment by gaby** [comment]
@corhere Correct, that's exactly what I was trying to do. I will close this since it seems outside the scope of `docker`. Thanks for the help
